Alloy/MTLSharedGraphicsBuffer.swift

```diff
@@ -0,0 +1,125 @@
+import Foundation
+import Alloy
+
+@available(iOS 12.0, *)
+final public class MTLSharedGraphicsBuffer {
+
+    public enum SupportedPixelFormat {
+        case bgra8Unorm
+
+        var mtlPixelFormat: MTLPixelFormat {
+            switch self {
+            case .bgra8Unorm:
+                return .bgra8Unorm
+            }
+        }
+    }
+
+    public enum Error: Swift.Error {
+        case initializationFailed
+    }
+
+    public let texture: MTLTexture
+    public let pixelBuffer: CVPixelBuffer
+    public let buffer: MTLBuffer
+    public var rect: CGRect {
+        let size = CGSize(width: self.texture.width,
+                          height: self.texture.height)
+        return .init(origin: .zero,
+                     size: size)
+    }
+    public var region: MTLRegion {
+        self.texture.region
+    }
+    
+    public var cgImage: CGImage {
+        get { self.cgContext.makeImage()! }
+        set { self.cgContext.draw(newValue, in: self.rect) }
+    }
+    public var image: UIImage {
```

Comment:
we are trying to avoid UIKit depdendencies inside alloy, I don't think it is necesasry